<?xml version="1.0"?>
<launch>
	<arg name="use_rviz" default="true" doc="if true, launch rviz" />
	<param name="left_wheel_joint" value="wheel_left_joint"/>
	<param name="right_wheel_joint" value="wheel_right_joint"/>
	<param name="vx_noise" value="0.01"/>
	<param name="w_noise" value="0.01"/>
	<param name="slip_max" value="0.4"/>
	<rosparam param="x_tubes">[2, -2, -0.5, 1, 0.6]</rosparam>
	<rosparam param="y_tubes">[2, 2.5, 0.2, 0.8, -0.3]</rosparam>
	<!-- "tube_var" is ordered as [sigma_x, sigma_y, sigma_xy] -->
	<rosparam param="tube_var">[0.03, 0.03, 0.01]</rosparam>
	<param name="tube_radius" value="0.0762"/>
	<param name="d_max_tubes" value="4.0"/>
	<param name="robot_radius" value ="0.105"/>  
	<rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />
	
	<node pkg="turtlebot3_teleop" type="turtlebot3_teleop_key" name="turtlebot3_teleop_keyboard" output="screen"/>
	<node pkg="nurtlesim" type="tube_world" name="tube_world" output="screen"/>
	<group if="$(arg use_rviz)">
		<node name="rviz" pkg="rviz" type="rviz" args="-d '$(find nurtlesim)/config/sim.rviz'"/>
	</group>
</launch>
